MBSD-369 MW upgrade to 1.43.5

wikivisor opened the pull request.

DanielButAtWork committed:
Stop using `MW_LOAD_EXTENSIONS` and `MW_LOAD_SKINS` (#301)

Load extensions previously loaded by `MW_LOAD_EXTENSIONS` manually in
LocalSettings.php

* The settings `$wgScribuntoUseGeSHi` and `$wgScribuntoUseCodeEditor` were
previously set based on whether the syntax highlighting and CodeEditor
extensions were enabled via `MW_LOAD_EXTENSIONS`; now that they are enabled
manually, set both configuration values to true.
* Local testing confirms that all of the extensions can be loaded via
`wfLoadExtension()` rather than from a PHP entrypoint.

Load skins previously loaded by `MW_LOAD_SKINS` manually in LocalSettings.php

* Taqasta automatically enabled the `Bootstrap` extension when the `chameleon`
skin was loaded via `MW_LOAD_SKINS`; now that the skin is manually loaded, the
extension needs to be manually loaded as well.
* Taqasta automatically sets `$wgDefaultSkin` to `Vector` when `MW_LOAD_SKINS`
is not used, even in the presence of of `MW_DEFAULT_SKIN`. Accordingly,
manually set `$wgDefaultSkin` back to `chameleon`. To avoid confusion about the
default skin being set in two places, also remove `MW_DEFAULT_SKIN`.

MBSD-375

wikivisor committed:
MBSD-369 Updates Taqasta

wikivisor committed:
MBSD-369 Updates Taqasta

wikivisor committed:
MBSD-369 Updates SemanticScribunto

wikivisor committed:
MBSD-369 Stops ContributionScores from looking into the retired ipblocks (bug T379715)

wikivisor committed:
MBSD-369 Enables Widgets

wikivisor committed:
MBSD-369 Enables Widgets

wikivisor committed:
MBSD-369 Enables Widgets

wikivisor committed:
MBSD-369 Removes dev mount of ContributionScores

wikivisor committed:
MBSD-369 Installs ExternalData

wikivisor committed:
MBSD-369 Installs ExternalData

wikivisor committed:
MBSD-369 Updates Taqasta to 1.43.5-20251202-01333b3

DanielButAtWork reviewed (changes requested):
instead of changing the configuration, we should update Taqasta to have the latest version of ContributionScores, just like I did in Canasta - its fixed in the latest commit on master

DanielButAtWork committed:
Install and configure SpamBlacklist (#303)

Actual addition of any email domains will need to be done on wiki, it could be
done in configuration but on-wiki management is clearer.

MBSD-378

DanielButAtWork committed:
Install and configure SpamBlacklist (#303)

Actual addition of any email domains will need to be done on wiki, it could be
done in configuration but on-wiki management is clearer.

MBSD-378

wikivisor committed:
MBSD-369 Updates Taqasta to 1.43.5-20251219-96a65ea

pastakhov committed:
Set innodb_buffer_pool_instances = 2 in mysql (it was 4) (#290)

wikivisor committed:
WLDR-382 Enables ExternalData

wikivisor committed:
WLDR-382 Enables ExternalData

wikivisor committed:
WLDR-382 Enables ExternalData

DanielButAtWork committed:
matomo: remove deprecated `forceSlash` Traefik setting (#308)

MBSD-384

wikivisor committed:
WLDR-382 Enables MagicNoCache

wikivisor committed:
WLDR-382 Ending new line fix

DanielButAtWork committed:
Update SemanticDepdencyUpdater to v4.2.0 (#318)

To include gesinn-it-pub/SemanticDependencyUpdater#17 and other improvements

MBSD-395

DanielButAtWork committed:
Store restic backups locally (#319)

Use the default location `/mnt/restic` which is already persisted as a docker
volume `restic_data`. Remove the AWS credentials and the configuration of the
restic repository.

WIK-2373

wikivisor committed:
Enable $smwgEnabledQueryDependencyLinksStore

pastakhov committed:
Force cache for main page in varnish (#321)

* use ( req.http.Cookie ~ "_session" && req.http.Cookie ~ "UserID" )

pastakhov committed:
Set innodb_buffer_pool_size = 1G (from 2G), add max-binlog-size=268435456 (#322)

DanielButAtWork committed:
Override SMW updateQueryDependencies to use job queue (#323)

Changes are noted inline with "WIKITEQ PATCH" comments

MBSD-387

DanielButAtWork committed:
Add UI images and support for SMTP server configuration (#324)

* Adds support for SMTP server configuration
* Adds UI images to the chameleon resources
---------
Co-authored-by: Vedmaka <god.vedmaka@gmail.com>

pastakhov committed:
Varnish: synthesize 405 for unknown HTTP methods (#325)

Replace pipe with edge 405; remove vcl_pipe. Allow GET, HEAD, PUT, POST, PATCH, DELETE, OPTIONS only.

OPTIONS: CORS preflight for cross-origin requests to MediaWiki API and REST (/w/rest.php/...).

PUT, PATCH, DELETE: used by some MediaWiki REST routes (/w/rest.php/...).

wikivisor committed:
Update Taqasta container to version 1.43.8 in compose.yml.
